oteljava/all/src/main/scala/org/typelevel/otel4s/oteljava/OtelJava.scala

```diff
@@ -88,6 +89,16 @@ object OtelJava {
     )
   }
 
+  /** Creates a no-op implementation of the [[OtelJava]].
+    */
+  def noop[F[_]: Applicative: LocalContext]: OtelJava[F] =
```

Comment:
It's a bit odd that we require `LocalContext` for no-op implementation. But it's required by the constructor.

Reply:
The following kinda solves the problem, but I have mixed feelings :) 

```scala
def noop[F[_]: Applicative]: OtelJava[F] = {
  val noopLocal: LocalContext[F] = new cats.mtl.Local[F, Context] {
    def local[A](fa: F[A])(f: Context => Context): F[A] = fa
    def applicative: Applicative[F] = Applicative[F]
    def ask[E2 >: Context]: F[E2] = Applicative[F].pure(Context.root)
  }
  
  new OtelJava(
    JOpenTelemetry.noop(),
    ContextPropagators.noop,
    MeterProvider.noop,
    TracerProvider.noop
  )(noopLocal)
}
```

Reply:
I have (I suspect the same) mixed feelings as well: it feels like it violates the contract of `Local`